### PR TITLE
Automated cherry pick of #128182: Fix crash on kube manager's service-lb-controller after v1.31.0.

### PR DESCRIPTION
**Pull request by carlory**
Cherry pick of #128182 on release-1.31.

#128182: Fix crash on kube manager's service-lb-controller after v1.31.0.

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Fixes 1.31 regression that can crash kube-controller-manager's service-lb-controller loop
```

### CODE REVIEW

**Comment by aojea** [comment]
it is better to squash it in one commit, I added the command on the CI to do it in the upstream PR , so it was merged as 442183a9298e37cb655b4a3f48462ee30e475069.

**Comment by carlory** [comment]
@aojea done.

**Comment by aojea** [comment]
/lgtm
/approve

Thanks

/kind bug
/kind regression

**Comment by aojea** [comment]
ping @kubernetes/release-managers 

/assign @cheftako

**Comment by aojea** [comment]
/assign @thockin  

I think you have approval rights on those folders

**Comment by jeremyrickard** [comment]
> ping @kubernetes/release-managers
> 
> /assign @cheftako

I will review this once it's been approved!

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
